From 9d15f07adb0de7ae02111f942bf7d0957bbc25d0 Mon Sep 17 00:00:00 2001
From: Pavlo Yatsukhnenko <yatsukhnenko@gmail.com>
Date: Fri, 20 Oct 2017 16:37:11 +0300
Subject: [PATCH] Issue #1224

---
 redis.c         | 2 ++
 redis_array.c   | 2 ++
 redis_cluster.c | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/redis.c b/redis.c
index ff83159f..18e7ef90 100644
--- a/redis.c
+++ b/redis.c
@@ -547,6 +547,8 @@ create_redis_object(zend_class_entry *ce TSRMLS_DC)
     zval *tmp;
     zend_hash_copy(redis->std.properties, &ce->default_properties,
         (copy_ctor_func_t)zval_add_ref, (void *)&tmp, sizeof(zval *));
+#else
+    object_properties_init(&redis->std, ce);
 #endif
 
     retval.handle = zend_objects_store_put(redis,
diff --git a/redis_array.c b/redis_array.c
index b293253f..01a48e5f 100644
--- a/redis_array.c
+++ b/redis_array.c
@@ -187,6 +187,8 @@ create_redis_array_object(zend_class_entry *ce TSRMLS_DC)
     zval *tmp;
     zend_hash_copy(obj->std.properties, &ce->default_properties,
         (copy_ctor_func_t)zval_add_ref, (void *)&tmp, sizeof(zval *));
+#else
+    object_properties_init(&obj->std, ce);
 #endif
 
     retval.handle = zend_objects_store_put(obj,
diff --git a/redis_cluster.c b/redis_cluster.c
index 94b0077c..f1136471 100644
--- a/redis_cluster.c
+++ b/redis_cluster.c
@@ -332,6 +332,8 @@ create_cluster_context(zend_class_entry *class_type TSRMLS_DC) {
 
     zend_hash_copy(cluster->std.properties, &class_type->default_properties,
         (copy_ctor_func_t)zval_add_ref, (void*)&tmp, sizeof(zval*));
+#else
+    object_properties_init(&cluster->std, class_type);
 #endif
 
     retval.handle = zend_objects_store_put(cluster,
